src/core/actor/DipolarBarnesHut.hpp

```diff
@@ -30,6 +30,9 @@
 #include "errorhandling.hpp"
 #include "grid.hpp"
 
+#include <cstdio>
```

Comment:
I think the method should not directly write output in the first place. Since `errorhandling` is already included, I think it would be better to just replace the printing by a runtime error. `iostream` is a _very_ large header, so reexporting has a negative impact on compile time. Same for direct sum...

Reply:
I'll have a look.